ClientFactory/GenericClientFactory.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Http\HttplugBundle\ClientFactory;
+
+use Http\Client\HttpClient;
+
+/**
+ * This is a client factory that will help you instantiate any client.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class GenericClientFactory
```

Comment:
why not use a service name pattern within the DI extension to use the correct factory service directly? does this level of indirection gain us anything additional?